eel/eel-background.c

```diff
@@ -665,8 +665,15 @@ widget_realized_setup (GtkWidget     *widget,
     self->details->screen_monitors_handler =
         g_signal_connect (screen, "monitors-changed", G_CALLBACK (screen_size_changed), self);
 
-    self->details->use_common_surface =
-        (gdk_window_get_visual (window) == gtk_widget_get_visual (widget)) ? TRUE : FALSE;
+    if  (GDK_IS_X11_DISPLAY (display))
+    {
+        GdkWindow *window;
+        window = gdk_screen_get_root_window (screen);
+        self->details->use_common_surface =
+            (gdk_window_get_visual (window) == gtk_widget_get_visual (widget)) ? TRUE : FALSE;
+    }
+    else /*Wayland is always composited*/
+         self->details->use_common_surface = TRUE;
```

Comment:
Indentation, 4 spaces for the else clause.

Reply:
Four spaces between else and the comment is indeed used elsewhere in the same file, adding it here

As for the indentation of the following line, here is an example of the following line from fr-command-unarchiver.c in engrampa, where the code uses tabs but includes another single-line else clause:
```
	if (overwrite)
		fr_process_add_arg (comm->process, "-f");
	else
		fr_process_add_arg (comm->process, "-s");
```
However, in this case that does in fact look odd, but indenting only four spaces might be misleading. Adding brackets for that, but can do something different if you prefer

Reply:
I do not understand why using brackets for the else clause wit one line, but ok.
Indentation is fixed.